konduit-serving-pipeline/src/main/java/ai/konduit/serving/pipeline/impl/data/helpers/ProtobufUtils.java

```diff
@@ -126,14 +261,74 @@ public static Data convertProtobufToData(DataProtoMessage.DataMap dataMap) {
             if (item.getTypeValue() == DataProtoMessage.DataScheme.ValueType.DOUBLE.ordinal()) {
                 retData.put(entry.getKey(), item.getDoubleValue());
             }
+            if (item.getTypeValue() == DataProtoMessage.DataScheme.ValueType.DATA.ordinal()) {
+                DataProtoMessage.DataMap itemMetaData = item.getMetaData();
+                Data embeddedData = deserialize(itemMetaData);
+                retData.put(entry.getKey(), embeddedData);
+            }
+            if (item.getTypeValue() == DataProtoMessage.DataScheme.ValueType.BOUNDING_BOX.ordinal()) {
+                DataProtoMessage.BoundingBox pbBox = item.getBoxValue();
+                BoundingBox boundingBox = null;
+                if (pbBox.getType() == DataProtoMessage.BoundingBox.BoxType.CHW)
+                    boundingBox = new BBoxCHW(pbBox.getCx(), pbBox.getCy(), pbBox.getH(), pbBox.getW());
+                else if (pbBox.getType() == DataProtoMessage.BoundingBox.BoxType.XY)
+                    boundingBox = new BBoxXY(pbBox.getX0(), pbBox.getX1(), pbBox.getY0(), pbBox.getY1());
```

Comment:
Missing the probabability and label fields here.

Reply:
Fixed.